src/fuzz_target.cc

```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "riscv.h"
+
+const int max_instructions = 5000;
+
+// In order to be able to inspect a coredump we want to
+// crash on every ASAN error.
+extern "C" void __asan_on_error()
```

Comment:
Why did you specify this file as C++ source instead of C?

Reply:
I followed [the steps](https://llvm.org/docs/LibFuzzer.html) on the official site, and it uses C++ code. 

From the stacktrace on error, I believe that the LibFuzzer is implemented in C++. 

```
#0 0x55a9931f3b21 in __sanitizer_print_stack_trace (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0x19eb21) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #1 0x55a9931663b8 in fuzzer::PrintStackTrace() (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0x1113b8) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #2 0x55a99314bda9 in fuzzer::Fuzzer::AlarmCallback() (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xf6da9) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #3 0x7f9c66c[42](https://github.com/sysprog21/rv32emu/actions/runs/6905735139/job/18789172120#step:4:43)51f  (/lib/x86_64-linux-gnu/libc.so.6+0x4251f) (BuildId: a[43](https://github.com/sysprog21/rv32emu/actions/runs/6905735139/job/18789172120#step:4:44)bfc8428df6623cd498c9c0caeb91aec9be4f9)
    #4 0x55a99316499a in __sanitizer_cov_trace_const_cmp1 (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0x10f99a) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #5 0x55a993273a78 in do_csw /home/runner/work/rv32emu/rv32emu/src/rv32_template.c:852:1
    #6 0x55a99323db7d in rv_step /home/runner/work/rv32emu/rv32emu/src/emulate.c:1029:13
    #7 0x55a9932c69a2 in run /home/runner/work/rv32emu/rv32emu/src/main.c:89:5
    #8 0x55a9932c69a2 in rv_init_and_execute_binary /home/runner/work/rv32emu/rv32emu/src/main.c:253:5
    #9 0x55a99322[47](https://github.com/sysprog21/rv32emu/actions/runs/6905735139/job/18789172120#step:4:48)df in fuzz_elf_loader(unsigned char const*, unsigned long) /home/runner/work/rv32emu/rv32emu/src/fuzz_target.cc:20:15
    #10 0x55a9932247df in LLVMFuzzerTestOneInput /home/runner/work/rv32emu/rv32emu/src/fuzz_target.cc:30:5
    #11 0x55a99314d3c3 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xf83c3) (BuildId: def03160f034cb06e8fb39819143fae47fa8a6[49](https://github.com/sysprog21/rv32emu/actions/runs/6905735139/job/18789172120#step:4:50))
    #12 0x55a99314cb19 in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool, bool*) (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xf7b19) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #13 0x55a99314e788 in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::vector<fuzzer::SizedFile, std::allocator<fuzzer::SizedFile> >&) (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xf9788) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #14 0x55a99314ec72 in fuzzer::Fuzzer::Loop(std::vector<fuzzer::SizedFile, std::allocator<fuzzer::SizedFile> >&) (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xf9c72) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #15 0x55a99313cfc2 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xe7fc2) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #16 0x[55](https://github.com/sysprog21/rv32emu/actions/runs/6905735139/job/18789172120#step:4:56)a993166cb2 in main (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0x111cb2) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
    #17 0x7f9c66c29d8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f) (BuildId: a43bfc8428df6623cd498c9c0caeb91aec9be4f9)
    #18 0x7f9c66c29e3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f) (BuildId: a43bfc8428df6623cd498c9c0caeb91aec9be4f9)
    #19 0x55a993131a04 in _start (/home/runner/work/rv32emu/rv32emu/build/rv32emu_fuzz+0xdca04) (BuildId: def03160f034cb06e8fb39819143fae47fa8a649)
```